book/scheduling-and-threading.md

```diff
@@ -93,137 +44,130 @@ class Task:
+to it than that. Before this change, we no choice but to run scripts
+right away just as they were loaded. But now that running scripts is a
+`Task`, the task runner controls when it runs. It could run only one
+script per second, or only when the page is active, or only if there
+isn't a higher-priority user action to respond to. A browser could
+even have multiple task runners, optimized for different use cases.
 
 Timers and setTimeout
-====================
+=====================
 
-It's often the case that when adding a task, it's not to be run right away, but
-instead at some point in the future. One example is the
-[`setTimeout`][settimeout] JavaScript API, which provides a way to run a
-function a given number of milliseconds from now. For example,
+Tasks are a natural way to support several JavaScript APIs that ask
+for a function to be run at some point in the future. For example, the
+[`setTimeout`][settimeout] JavaScript API lets you run a function some
+number of milliseconds from now. This code prints "Callback" to the
+console one minute in the future, for example:
+
+[settimeout]: https://developer.mozilla.org/en-US/docs/Web/API/setTimeout
 
 ``` {.javascript expected=false}
-function callback() {
-    console.log('Callback')
-}
+function callback() { console.log('Callback'); }
 setTimeout(callback, 1000);
 ```
-will print "Callback" to the console log one second from now.
 
-This API *could* be implemented by recording a time associated with each new
-`Task`, and comparing that time against the current time in the event
-loop.^[This approach is called
-*polling*, and is also what the SDL event loop does to look for events and
- tasks.] A better approach is to use Python's [`threading.Timer`][timer] class,
- which does this for you (and probably much more efficiently) with a
- [Python thread][python-thread].
+We can implement `setTimeout` in our browser using the
+[`Timer`][timer] class in Python's [`threading`][threading] module.
+You use the class like this:
 
 [timer]: https://docs.python.org/3/library/threading.html#timer-objects
-[settimeout]: https://developer.mozilla.org/en-US/docs/Web/API/setTimeout
-[python-thread]: https://docs.python.org/3/library/threading.html
+[threading]: https://docs.python.org/3/library/threading.html
 
-Start by importing that module:
-
-``` {.python}
+``` {.python expected=False}
 import threading
+threading.Timer(1, callback).start()
 ```
 
-The `Timer` class lets you run a callback at a specified time in the future. It
-takes two parameters: a time delta in seconds from now, and a function to call
-when that time expires. The following code will run `callback` 10 seconds in
-the future on a new Python thread:
-
-``` {.python expected=False}
-threading.Timer(10, callback).start()
-```
+This runs `callback` in 1 seconds on a new Python thread. Now, it's
+going to be a little tricky to use `Timer` to implement `setTimeout`
+due to the fact that multiple threads will be involved.
```

Comment:
```suggestion
due to the fact that multiple threads will be involved, but it's worth it.
```

Reply:
This again is related to the polling sentence you deleted and my comment on it..

Reply:
This is good whether or not polling is presented first!